README.md

```diff
@@ -18,83 +18,77 @@ import xcomponentapi from 'reactivexcomponent.js';
+                const stateMachineUpdateListener = (jsonData) => {
                     //jsonMessage property is the public member
                     console.log(jsonData.jsonMessage);
                     //send context using directly stateMachineRef
-                    jsonData.stateMachineRef.send(messageType3, jsonMessage3);
+                    jsonData.stateMachineRef.send(messageType, jsonMessage);
                 }       
                 // subscribe using a callback                
                 subscriber.subscribe(componentName, stateMachineResponse, stateMachineUpdateListener);  
-                // subscribe using a Promise
-                subscriber.subscribePromise(componentName, stateMachineResponse).then(jsonData => stateMachineUpdateListener(jsonData));
             }
 
-            var snapshotListener = (items) => {
-                //items is an array of instances of stateMachineResponse                
+            //Snapshot using with a Promise
+            subscriber.getSnapshot(componentName, stateMachineResponse).then(items => {
                 items.forEach(item => {
                     console.log(item.jsonMessage);
                     //each item contains stateMachineRef with a send method to publish an event to an instance                    
                     item.stateMachineRef.send(messageType, jsonMessage);
                 })
-            };
-
-            //Snapshot with a callback
-            subscriber.getSnapshot(componentName, stateMachineResponse, (items) => {
-                snapshotListener(items);
             });
 
-            // Snapshot using with a Promise
-            subscriber.getSnapshotPromise(componentName, stateMachineResponse).then(items => {
-                snapshotListener(items);
-            });
-
-            //create a publisher to send an event
-            var publisher = session.createPublisher(); 
+            //Create a publisher to send an event
+            const publisher = session.createPublisher(); 
 
-            //check if publisher of stateMachineName is exposed by xcApi
+            //Check if publisher of stateMachineName is exposed by xcApi
             if (publisher.canPublish(componentName, stateMachineName, messageType1)) {
                 //visibility parameter is optional. It is false by default.
-                publisher.send(componentName, stateMachineName, messageType1, jsonMessage1, visibilityPrivate);
+                publisher.send(componentName, stateMachineName, messageType1, jsonMessage, visibilityPrivate);
             } 
 
         }
 
-        var xcApiName = "HelloWorldApi.xcApi"            
-        xcomponentapi.createSession(xcApiName, serverUrl, sessionListener);
-        xcomponentapi.createSessionPromise(xcApiName, serverUrl).then(session => {
-            // use session object as specified before ...
+        // handle imprevisible session close using a Promise
+        xcomponentapi.closeSessionError(serverUrl)
```

Comment:
It is not an interesting use of promise (we are not dealing with an async action - it's just a notification).
https://stackoverflow.com/questions/45041462/node-js-when-to-use-promises-vs-callbacks
http://blog.millermedeiros.com/callbacks-promises-signals-and-events/
According to the name closeSessionError, it should be a property of the session... @maurelio1234